src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import duke.exception.ExceptionMessages;
+import duke.Ui;
+
+import java.util.ArrayList;
+
+public class TaskList {
+
+
+    private final ArrayList<Task> taskList;
+
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        taskList = tasks;
+    }
+
+
+    public void addTask(Task task) throws DukeException {
+        if (task.getDescription().equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_DESCRIPTION);
+        }
+        taskList.add(task);
+        final int listLength = taskList.size();
+        //To check that task was added successfully
+        final Task mostRecentTask = taskList.get(listLength - 1);
+        Ui.printAddTaskMessage(listLength, mostRecentTask);
+    }
+
+    public void addTaskWithoutMessage(Task task) throws DukeException {
+        if (task.getDescription().equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_DESCRIPTION);
+        }
+        taskList.add(task);
+    }
+
+
+    public void deleteTask(String input) throws DukeException, NumberFormatException {
+        if (input.equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_TASK_NUMBER);
+        }
+        int taskNumber = Integer.parseInt(input.trim());
+        if (isMissingTask(taskNumber)) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_INVALID_TASK_NUMBER);
+        }
+        deleteExistingTask(taskNumber);
+    }
+
+    private void deleteExistingTask(int taskNumber) {
+        final int taskIndex = taskNumber - 1;
+        Ui.printDeleteTaskSuccessMessage(taskList.get(taskIndex));
+        taskList.remove(taskIndex);
+    }
+
+
+    public void markTaskAsDone(String input) throws DukeException, NumberFormatException {
+        if (input.equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_TASK_NUMBER);
+        }
+        int taskNumber = Integer.parseInt(input.trim());
+        if (isMissingTask(taskNumber)) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_INVALID_TASK_NUMBER);
+        }
+        markExistingTaskAsDone(taskList.get(taskNumber - 1));
+    }
+
+    private boolean isMissingTask(int taskNumber) {
+        final int listLength = taskList.size();
+        return taskNumber <= 0 || taskNumber > listLength;
```

Comment:
I would suggest assigning a variable name for this compound condition to improve readability.